lib/graphql/auth-guards.ts

```diff
@@ -0,0 +1,16 @@
+import { Role } from '@prisma/client';
+import { Resolver } from './resolvers';
+
+export const authenticate =
+  (resolver: Resolver, authorizedRoles: Array<Role>): Resolver =>
+  (root, args, context, info) => {
+    const userRole = context.session?.user?.role;
+
+    // If user has the necessary permissions, allow the query or mutation to go through.
+    if (userRole && authorizedRoles.includes(userRole)) {
```

Comment:
We also want to give admins full access to the platform here

Reply:
Role.Admin is passed in every resolver that's wrapped so admins will always have access. I guess to remove redundancy we could just add an extra check here for admins instead of always passing them in as authorized roles. What do you think?

Reply:
Yeah, exactly what I was thinking!